server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java

```diff
@@ -461,7 +471,8 @@ public enum APIBlock implements Writeable {
         READ("read", INDEX_READ_BLOCK),
         WRITE("write", INDEX_WRITE_BLOCK),
         METADATA("metadata", INDEX_METADATA_BLOCK),
-        READ_ONLY_ALLOW_DELETE("read_only_allow_delete", INDEX_READ_ONLY_ALLOW_DELETE_BLOCK);
+        READ_ONLY_ALLOW_DELETE("read_only_allow_delete", INDEX_READ_ONLY_ALLOW_DELETE_BLOCK),
+        WRITE_ONLY_ALLOW_DELETE("write_only_allow_delete", INDEX_WRITE_ONLY_ALLOW_DELETE_BLOCK);
```

Comment:
The name of our blocks are currently inconsistent (e.g. `APIBlock.READ` blocks reads as one might expect, yet `APIBlock.READ_ONLY` blocks writes). However, making something named `WRITE_ONLY_ALLOW_DELETE` be functionally identical to something named `READ_ONLY_ALLOW_DELETE` doesn't seem to be helping the situation. Given the inconsistency already present, I don't have an obvious alternative suggestion, but can we try to work through a better name if we go forward with this approach?